File: 977.Square-of-a-Sorted-Array/square-of-a-sorted-array.js
// Given an integer array nums sorted in non-decreasing order, 
// return an array of the squares of each number sorted in non-decreasing order.

const sortedSquares = function(nums) {
    const sortedNums = nums.sort(function(a, b) {return a-b});
    for (let i = 0; i <= sortedNums.length; i++) {
        let squaredItem = sortedNums[i]*sortedNums[i];
        array[i] = squaredItem;
    }
    return array;
};

nums = [-7,-3,2,3,11]

File: 977.Square-of-a-Sorted-Array/square-of-a-sorted-array.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        if len(nums) < 0:
            return 0
        
        if len(nums) == 1:
            nums[0] = nums[0] * nums[0]
            return nums
        
        for i in range(len(nums)):
            temp = nums[i] * nums[i]
            nums[i] = temp
        
        nums.sort()
        return nums
    